services/app/src/components/views/quest/cardtemplates/decision/Actions.test.tsx

```diff
@@ -1,39 +1,212 @@
+    });
+    test('counts successes and ignores other rolls', () => {
+      expect(computeSuccesses([5, 20, 10, 5], {difficulty: 'medium'})).toEqual(1);
+    });
+    test('respects difficulty', () => {
+      expect(computeSuccesses([16, 15, 10, 14], {difficulty: 'hard'})).toEqual(0);
+    });
   });
   describe('computeOutcome', () => {
-    test.skip('computes success', () => { /* TODO */ });
-    test.skip('computes failure', () => { /* TODO */ });
-    test.skip('computes interrupted', () => { /* TODO */ });
-    test.skip('computes retry', () => { /* TODO */ });
-    test.skip('returns null when no selected decision', () => { /* TODO */ });
+    const selected = {difficulty: 'medium', requiredSuccesses: 5};
+
+    test('computes success', () => {
+      expect(computeOutcome([20, 20, 20, 20, 20], selected, TEST_SETTINGS, TEST_RP)).toEqual(Outcome.success);
+    });
+    test('computes failure', () => {
+      expect(computeOutcome([1], selected, TEST_SETTINGS, TEST_RP)).toEqual(Outcome.failure);
+    });
+    test('computes interrupted', () => {
+      expect(computeOutcome([20, 20, 20, 20, 10], selected, TEST_SETTINGS, TEST_RP)).toEqual(Outcome.interrupted);
+    });
+    test('computes retry', () => {
+      expect(computeOutcome([20, 20, 20, 20], selected, TEST_SETTINGS, TEST_RP)).toEqual(Outcome.retry);
+    });
+    test('returns null when no rolls', () => {
+      expect(computeOutcome([], selected, TEST_SETTINGS, TEST_RP)).toEqual(null);
+    });
   });
   describe('generateLeveledChecks', () => {
-    test.skip('returns 3 semi-unique, generated checks', () => { /* TODO */ });
-    test.skip('scales num successes with num adventurers', () => { /* TODO */ });
+    test('returns 3 semi-unique, generated checks', () => {
+      expect(generateLeveledChecks(5, seedrandom.alea('1234'))).toEqual([
+        {difficulty: 'hard', persona: 'light', requiredSuccesses: 1, skill: 'athletics'},
+        {difficulty: 'medium', persona: 'dark', requiredSuccesses: 3, skill: 'athletics'},
+        {difficulty: 'easy', persona: 'light', requiredSuccesses: 1, skill: 'knowledge'},
+      ]);
+    });
+    test('scales num successes with num adventurers', () => {
+      expect(Math.max(...generateLeveledChecks(1, seedrandom.alea('1234')).map((c) => c.requiredSuccesses))).toEqual(1);
+      expect(Math.max(...generateLeveledChecks(6, seedrandom.alea('1234')).map((c) => c.requiredSuccesses))).toBeGreaterThan(3);
+    });
     test.skip('scales difficulty with the number of times this type of check was selected previously', () => { /* TODO */ });
   });
   describe('skillTimeMillis', () => {
-    test.skip('respects settings', () => { /* TODO */ });
-    test.skip('scales with player count', () => { /* TODO */ });
+    test('scales with player count', () => {
+      expect(skillTimeMillis(TEST_SETTINGS, TEST_RP)).toBeLessThan(skillTimeMillis({...TEST_SETTINGS, numPlayers: 1}, initialMultiplayer));
```

Comment:
TEST_SETTINGS has 3 players.... so you have less time with more players? That's the opposite of scaling with player count, isn't it?

If anything, I would expect this to work the same as the combat timer, where 1 player gets 2x the time as everyone else, but the time doesn't change with the # of players (thereby giving smaller parties a slight communication advantage to compensate for their disadvantage in numbers)

Reply:
You've actually described exactly the test behavior... refactored to make it more clear.